### PR TITLE
Feature/S3C-3525 Create Tenant-1

### PR DESCRIPTION
**Pull request by saitharunthodupunuru**
Created this micro PR#1 for Create Tenant

Description:
* Renamed platform-admin-client to vault-admin-client
* Importing Vaultclient as a dependency from maven repo
* Publishing the jar artifact after successful build to OSS Sonatype Nexus Maven repository. As Apache Maven central repository only accepts release artifacts and the only way to publish an artifact to maven central is using open source OSS Nexus repository. We created account at https://oss.sonatype.org/#view-repositories and uploading gpg signed snapshot artifact before release.

References:

https://maven.apache.org/repository/guide-central-repository-upload.html
https://www.albertgao.xyz/2018/01/18/how-to-publish-artifact-to-maven-central-via-gradle/
https://central.sonatype.org/pages/ossrh-guide.html

### CODE REVIEW

File: upload-artifact.gradle [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apply plugin: 'com.bmuschko.nexus'
+apply plugin: 'signing'
+nexus {
+    sign = true
+    repositoryUrl = 'https://oss.sonatype.org/service/local/staging/deploy/maven2/'
+    snapshotRepositoryUrl = 'https://oss.sonatype.org/content/repositories/snapshots/'
+}
```
Comment (anurag4DSB) [inline upload-artifact.gradle:7]: missing new line

Comment (saitharunthodupunuru) [reply upload-artifact.gradle]: done

**Review by anurag4DSB — APPROVED** [review]

**Comment by anurag4DSB** [comment]
We need documentation as well, but can be handled in another PR.

**Review by ilkescality — APPROVED** [review]

**Pull request merged**